edge-agent/test/Microsoft.Azure.Devices.Edge.Agent.DiagnosticsComponent.Test/MetricsFileStorageTest.cs

```diff
@@ -0,0 +1,130 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+namespace Microsoft.Azure.Devices.Edge.Agent.DiagnosticsComponent.Test
+{
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Linq;
+    using Microsoft.Azure.Devices.Edge.Util;
+    using Microsoft.Azure.Devices.Edge.Util.Test.Common;
+    using Moq;
+    using Xunit;
+
+    public class MetricsFileStorageTest : IDisposable
+    {
+        DateTime fakeTime;
+        ISystemTime systemTime;
+        TempDirectory tempDirectory = new TempDirectory();
+
+        public MetricsFileStorageTest()
+        {
+            var systemTime = new Mock<ISystemTime>();
+            this.fakeTime = new DateTime(100000000);
+            systemTime.Setup(x => x.UtcNow).Returns(() => this.fakeTime);
+            this.systemTime = systemTime.Object;
+        }
+
+        [Fact]
+        public void Storage()
+        {
+            string directory = this.tempDirectory.GetTempDir();
+            MetricsFileStorage storage = new MetricsFileStorage(directory, this.systemTime);
+
+            storage.WriteData(string.Join(", ", Enumerable.Range(0, 10)));
+
+            Assert.NotEmpty(Directory.GetFiles(directory));
```

Comment:
you should also assert the file name and its content, right?

Reply:
This just checks that it makes a file. Subsequent tests validate it stores the correct data.

Reply:
It is odd, what this writeData method does is to write given data in a file named using current time.  therefore I think verify file name and content should be in this test.